sw/host/opentitanlib/src/spiflash/sfdp.rs

```diff
@@ -0,0 +1,529 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use byteorder::{LittleEndian, ReadBytesExt};
+use num_enum::FromPrimitive;
+use serde::Serialize;
+use std::convert::TryFrom;
+use thiserror::Error;
+
+use crate::util::bitfield::BitField;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("IO error: {0:?}")]
+    Io(std::io::Error),
+    #[error("the range {0}..{1} is out of bounds")]
+    SliceRange(usize, usize),
+}
+
+#[derive(Debug, Serialize)]
+/// The SFDP header identifies a valid SFDP, its version and the number of parameter headers.
+pub struct SfdpHeader {
+    pub signature: u32,
+    pub minor: u8,
+    pub major: u8,
+    pub nph: u8,
+    pub reserved: u8,
+}
+
+impl TryFrom<&[u8]> for SfdpHeader {
+    type Error = Error;
+    fn try_from(buf: &[u8]) -> Result<Self, Self::Error> {
+        let mut rdr = std::io::Cursor::new(buf);
+        Ok(SfdpHeader {
+            signature: rdr.read_u32::<LittleEndian>().map_err(|e| Error::Io(e))?,
```

Comment:
You can just use a function reference: `map_err(Error::Io)`. That said, if you're using `thiserror`, you can annotate `Io` with `#[from]`, which will allow you to just use `?` and let `Into` conversion do the trick.

Reply:
Done using the `#[from]` annotation.